webapi/src/main/scala/org/knora/webapi/responders/v1/ValueUtilV1.scala

```diff
@@ -120,6 +120,24 @@ class ValueUtilV1(private val settings: SettingsImpl) {
     }
 
     /**
+      * Creates a URL pointing to the given resource class icon. From the resource class Iri it gets the ontology specific path, i.e. the ontology name.
+      * If the resource class Iri is "http://www.knora.org/ontology/knora-base#Region", the ontology name would be "knora-base".
+      * To the base path, the icon name is appended. In case of a region with the icon name "region.gif",
+      * "http://salsahapp:port/project-icons-basepath/knora-base/region.gif" is returned.
+      *
+      * @param resourceClassIri the Iri of the resource class in question.
+      * @param iconsSrc the name of the icon file.
+      */
+    def makeResourceClassIconURL(resourceClassIri: IRI, iconsSrc: String): IRI = {
+        // get ontology name, e.g. "knora-base" from "http://www.knora.org/ontology/knora-base#Region"
+        // add +1 to ignore the slash
+        val ontologyName = resourceClassIri.substring(resourceClassIri.lastIndexOf('/') + 1, resourceClassIri.lastIndexOf('#'))
```

Comment:
Yes, I see the problem for other ontologies than ours.

I put this restriction in the function's comment. For our test data, this requirement is fulfilled. 
For the case of `FOAF` we could provide an ontology like knora-base:dc.